esprima.js

```diff
@@ -4658,7 +4758,11 @@
             }
         }
 
+        previousAllowYield = state.allowYield;
+        state.allowYield = false;
```

Comment:
This is not quite right. `yield` expression can be in the initializers of non-generator function parameters. By setting `allowYield` to false, we also allows `yield` as identifiers in generators parameters in non strict mode, which is banned by the spec.

Due to this problem, the `GeneratorParameter` flag cannot be omitted.